examples/README.md

```diff
@@ -0,0 +1,279 @@
+
+File: Makefile   
+```TOP = ..
+include $(TOP)/build.mk
+
+EXAMPLES = auction echo helloworld
+
+build clean:
+	$(foreach DIR, $(EXAMPLES), $(MAKE) -C $(DIR) $@ || exit;)
+```
+  
+Next step is to install, instantiate the chaincode and invoke transactions.  For this, 
+you use the integration framework.  Move to `FPC-INSTALL-DIR/integration`.   In the existing `Makefile`,  make these modifications: 
+
+```...
+test: auction_test echo_test deployment_test helloworld_test
+...
+helloworld_test:
+        ./helloworld_test.sh
+```        
+### Time to test !  
+Create a file to install, instantiate the chaincode, submit transactions.  The following test script invokes the transaction, putData thrice;  after each time, invokes getData to read the value set before. 
+In `FPC-INSTALL-DIR/integration` folder, Create file `helloworld_test.sh`
+```
+
+SCRIPTDIR="$(dirname $(readlink --canonicalize ${BASH_SOURCE}))"
+FPC_TOP_DIR="${SCRIPTDIR}/.."
+CONFIG_HOME="${SCRIPTDIR}/config"
+FABRIC_SCRIPTDIR="${FPC_TOP_DIR}/fabric/bin/"
+
+. ${FABRIC_SCRIPTDIR}/lib/common_utils.sh
+. ${FABRIC_SCRIPTDIR}/lib/common_ledger.sh
+
+CC_ID=helloworld_test
+
+#this is the path that will be used for the docker build of the chaincode enclave
+ENCLAVE_SO_PATH=examples/helloworld/_build/lib/
+
+CC_VERS=0
+num_rounds=3
+
+helloworld_test() {
+    # install, init, and register (auction) chaincode
+    try ${PEER_CMD} chaincode install -l fpc-c -n ${CC_ID} -v ${CC_VERS} -p ${ENCLAVE_SO_PATH}
+    sleep 3
+
+    try ${PEER_CMD} chaincode instantiate -o ${ORDERER_ADDR} -C ${CHAN_ID} -n ${CC_ID} -v ${CC_VERS} -c '{"args":["init"]}' -V ecc-vscc
+    sleep 3
+
+    try ${PEER_CMD} chaincode invoke -o ${ORDERER_ADDR} -C ${CHAN_ID} -n ${CC_ID} -c '{"Args":["setup", "ercc"]}' --waitForEvent
+
+    try ${PEER_CMD} chaincode query -o ${ORDERER_ADDR} -C ${CHAN_ID} -n ${CC_ID} -c '{"Args":["getEnclavePk"]}'
+
+    say "do hello world"
+    for (( i=1; i<=$num_rounds; i++ ))
+    do
+        # submit data $i for "asset1"
+        try ${PEER_CMD} chaincode invoke -o ${ORDERER_ADDR} -C ${CHAN_ID} -n ${CC_ID} -c '{"Args":["[\"putData\",\"asset1\",\"'$i'\"]", ""]}' --waitForEvent
+        
+	# get data for "asset1";  should be $i;
+        try ${PEER_CMD} chaincode invoke -o ${ORDERER_ADDR} -C ${CHAN_ID} -n ${CC_ID} -c '{"Args":["[\"getData\",\"asset1\"]", ""]}' --waitForEvent
+    done
+}
+
+# 1. prepare
+para
+say "Preparing Helloworld Test ..."
+# - clean up relevant docker images
+docker_clean ${ERCC_ID}
+docker_clean ${CC_ID}
+
+trap ledger_shutdown EXIT
+
+
+para
+say "Run helloworld  test"
+
+say "- setup ledger"
+ledger_init
+
+say "- helloworld test"
+helloworld_test
+
+say "- shutdown ledger"
+ledger_shutdown
+
+para
+yell "Helloworld test PASSED"
+
+exit 0
+```
+
+Execute:  
+```
+cd FPC-INSTALL-DIR/integration
+make helloworld_test
```

Comment:
again I have the feeling that it would be nice to have the following:
assuming we are still in $GOPATH/.../fabric-private-chaincode/examples/helloworld
```
./helloworld_test.sh
```